DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/ViewTreeSnapshot.swift

```diff
@@ -54,10 +56,10 @@ internal typealias Node = SessionReplayNode
 // An individual resource in `ViewTreeSnapshot`. It is used to describe binary representation of heavy resources such as images.
 @_spi(Internal)
 public protocol SessionReplayResource {
-    /// The unique identifier of the resource.
-    var identifier: String { get }
-    /// The data of the resource.
-    var data: Data { get }
+    /// Calculates the unique identifier of the resource.
+    func calculateIdentifier() -> String
+    /// Calculates the data of the resource.
+    func calculateData() -> Data
```

Comment:
flag/ These APIs are not thread safe as their implementation works on shared state. We could at least describe this in API comment, otherwise we risk a crash.